src/main/scala/com/mozilla/telemetry/mozdata/Utils.scala

```diff
@@ -0,0 +1,115 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+package com.mozilla.telemetry.mozdata
+
+import org.apache.log4j.Logger
+import org.apache.spark.sql.SparkSession
+import org.apache.spark.sql.catalyst.analysis.NoSuchTableException
+import org.json4s.{DefaultFormats, Formats}
+import org.json4s.jackson.Serialization.write
+
+import scala.collection.immutable.ListMap
+import scala.io.Source
+import scala.util.matching.Regex
+
+object Utils {
+  val locationRegex: Regex = "(?s).*LOCATION[^']+'([^']+)'.*".r
+
+  /** Update metadata for table */
+  def updateMetadata(sqlTableName: String, method: String,
+                     spark: SparkSession): Unit = method match {
+    case "sql_repair" => spark.sql(s"MSCK REPAIR TABLE `$sqlTableName`")
+    case "sql_refresh" => spark.sql(s"REFRESH TABLE `$sqlTableName`")
+    // TODO support "lambda" method
+    // TODO support "glue" method
+    case value => throw new IllegalArgumentException(s"Unsupported metadata location: $value"
+    )
+  }
+
+  /** check if version is in MozData version format */
+  def isVersion(version: String): Boolean = version.matches("v[0-9]+")
+
+  /** try to get table version from uri */
+  def uriVersion(uri: Option[String]): Option[String] = {
+    uri.getOrElse("").split("/").last match {
+      case version if isVersion(version) => Some(version)
+      case _ => None
+    }
+  }
+
+  /** Extract non-catalog table info
+    *
+    * reusable code for getTableInfo
+    *
+    * @param tableUri uri of table without version
+    * @param version optional specific version of table
+    * @return same as getTableInfo
+    */
+  def getFsTable(tableUri: String, version: Option[String]):
+  (Option[String], Option[String], Option[String]) = {
+    val detectedVersion = version.getOrElse({
+      val versions = Hadoop.ls(tableUri)._1.filter(isVersion)
+      if (versions.isEmpty) {"v1"} // new table, fall back to version "v1"
+      else {versions.maxBy(_.substring(1).toInt)} // use latest version
+    })
+    (Some(s"$tableUri/$detectedVersion"), Some(detectedVersion), None)
+  }
+
+  /** Locate information needed for accessing the given table
+    *
+    * @param tableName table to look up
+    * @param version optional specific version of table
+    * @param owner optional email that identifies non-global namespace
+    * @param uri optional non-standard location for this table
+    * @param spark spark session used to access data
+    * @param adHocTablesDir filesystem location of ad hoc tables
+    * @param globalTablesDir filesystem location of global tables
+    * @return (detectedUri, detectedVersion, sqlTableName) of table, at
```

Comment:
It would be good to document here when we return just a `sqlTableName`, just a `detectedUri`, and when we return one or the other with `detectedVersion`. I find myself needing to reference that information pretty consistently elsewhere.